dotCMS/src/main/java/com/dotmarketing/portlets/workflows/util/WorkflowActionletUtil.java

```diff
@@ -0,0 +1,181 @@
+                                    false);
+                    userSet.add(user);
+                } catch (Exception e) {
+                    // Email not found
+                    idNotFound = Boolean.TRUE;
+                    exception = e;
+                }
+            } else {
+                try {
+                    User user = APILocator.getUserAPI()
+                            .loadUserById(token, APILocator.getUserAPI().getSystemUser(), false);
+                    userSet.add(user);
+                    continue;
+                } catch (Exception e) {
+                    // User ID not found
+                    idNotFound = Boolean.TRUE;
+                    exception = e;
+                }
+                try {
+                    final Role role = APILocator.getRoleAPI().loadRoleByKey(token);
+                    final List<User> approvingUsersInRole = APILocator.getRoleAPI()
+                            .findUsersForRole(role);
+                    userSet.addAll(approvingUsersInRole);
+                    idNotFound = Boolean.FALSE;
+                } catch (DotSecurityException e) {
+                    Logger.warn(WorkflowActionletUtil.class,
+                            "An error occurred when retrieving users from role key: " + token,
+                            e);
+                } catch (DotDataException e) {
+                    // The specified role could not be found
+                    idNotFound = Boolean.TRUE;
+                    exception = e;
+                }
+            }
+            if (idNotFound) {
+                Logger.warn(WorkflowActionletUtil.class,
+                        "The following email/userID/role key could not be found: " + token,
+                        exception);
+            }
+        }
+        return userSet;
+    }
+
+    /**
+     * Returns the value of the specified actionlet parameter. If the value is {@code null}, the
+     * specified default value will be returned.
+     *
+     * @param parameter    The actionlet parameter specified through the UI.
+     * @param defaultValue The default value of the parameter if null.
+     *
+     * @return The value of the specified parameter.
+     */
+    public static <T> T getParameterValue(final WorkflowActionClassParameter parameter,
+            final T defaultValue) {
+        final Object paramValue = getParameterValue(parameter);
+        return (null != paramValue) ? (T) paramValue : defaultValue;
+    }
+
+    /**
+     * Returns the value of the specified actionlet parameter.
+     *
+     * @param parameter The actionlet parameter specified through the UI.
+     *
+     * @return The value of the specified parameter.
+     */
+    public static <T> T getParameterValue(final WorkflowActionClassParameter parameter) {
+        T paramValue = null;
+        try {
+            paramValue = (T) parameter.getValue();
+        } catch (Exception e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)

Reply:
let's log something here